storage/client/queries/queries.go

```diff
@@ -286,31 +286,82 @@ const (
 			ORDER BY id DESC`
 
 	ValidatorsData = `
+		WITH
+		-- Find all self-delegations for all accounts with active delegations.
+		self_delegations AS (
+			SELECT delegatee AS address, shares AS shares
+				FROM chain.delegations
+				WHERE delegator = delegatee
+				GROUP BY delegator, delegatee
+		),
+		-- Compute the number of delegators for each validator.
+		delegators_count AS (
+			SELECT delegatee AS address, COUNT(*) AS count
+				FROM chain.delegations
+				GROUP BY delegatee
+		),
+		-- Each entity can have multiple nodes. For each entity, find the validator node with the highest voting power.
```

Comment:
An additional note here would be nice: only one of the entity nodes will have non-zero voting power.